bin/pybsn-repl

```diff
@@ -26,15 +26,26 @@ import argparse
 import logging
 import textwrap
 import re
+import os
 from traitlets.config.loader import Config
 from IPython.terminal.ipapp import TerminalIPythonApp
 from IPython.core.inputtransformer import StatelessInputTransformer
 from IPython.core.magic import Magics, magics_class, line_magic
 
+
+def env_var(value):
+    if value not in os.environ:
+        raise argparse.ArgumentTypeError("'%s' is not an environment variable" % value)
+    return os.environ[value]
+
+
 parser = argparse.ArgumentParser(description=__doc__)
 
 parser.add_argument('--host', '-H', type=str, default="127.0.0.1", help="Controller IP/Hostname to connect to")
-parser.add_argument('--token', '-t', type=str, help="Session/Token to use")
+token_source = parser.add_mutually_exclusive_group()
+token_source.add_argument('--token', '-t', type=str, help="Session/Token to use")
+token_source.add_argument('--env-token', '-e', type=env_var, dest='token',
+                          help="Environment variable of session/token to use")
```

Comment:
Would it work to just always take `$FL_SESSION_COOKIE` from env (if present) and override that with `--token` from the command line (if specified)?

Reply:
by default it is currently trying to auth with `user`:`password`, I didn't want to change the default behavior to attempt to use a token instead. worth a thought though.

Reply:
cool, providing a function in `type=` makes it be called to parse the argument? Didn't know this...